stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -650,17 +666,30 @@ extension DistributedActorSystem {
       if returnType == Void.self {
         try await handler.onReturnVoid()
       } else {
-        func invokeOnReturn<R>(_ returnType: R.Type) async throws {
-          let value = resultBuffer.load(as: returnType)
-          try await handler.onReturn(value: value)
-        }
-
-        try await _openExistential(returnType, do: invokeOnReturn)
+        try await self.invokeHandlerOnReturn(
+          handler: handler,
+          resultBuffer: resultBuffer,
+          metatype: returnType
+        )
       }
     } catch {
       try await handler.onThrow(error: error)
     }
   }
+
+  @_alwaysEmitIntoClient
+  public func invokeHandlerOnReturn(
```

Comment:
@slavapestov should I add this into a separate extension that is `@available(SwiftStdlib 5.11, *)`?

Reply:
I think this goes back to the question about availability of new requirements because if we mark `onReturn` as conditionally available we'd have to match that availability here as well

Reply:
This was a dead-end because `onReturn` is a new requirement.